mne/time_frequency/tests/test_psd.py

```diff
@@ -13,102 +15,193 @@
+    psds_proj, freqs_proj = psd_multitaper(epochs[:1], proj=True, **kws_mt)
 
-    assert_array_almost_equal(psds, psds_proj)
-    assert_true(psds.shape == (1, len(picks), len(freqs)))
+    # this one will fail if you add for example 0.1 to tmin
+    assert_array_almost_equal(psds, psds_full, 27)
+    assert_array_almost_equal(psds, psds_proj, 27)
+
+    assert_true(psds.shape == (1, len(picks_mt), len(freqs)))
     assert_true(np.sum(freqs < 0) == 0)
     assert_true(np.sum(psds < 0) == 0)
 
 
+@requires_version('scipy', '0.12')
+def test_psd_epochs_deprecate():
+    """Test PSD estimation on epochs
+    """
+    with warnings.catch_warnings(record=True):
+        warnings.simplefilter('always')
+        raw = io.Raw(raw_fname)
+
+        exclude = raw.info['bads'] + ['MEG 2443', 'EEG 053']  # bads + 2 more
+
+        # picks MEG gradiometers
+        picks = pick_types(raw.info, meg='mag', eeg=False, stim=False,
+                           exclude=exclude)
+
+        picks = picks[:2]
+
+        n_fft = 512  # the FFT size (n_fft). Ideally a power of 2
+
+        tmin, tmax, event_id = -0.5, 0.5, 1
+        include = []
+        raw.info['bads'] += ['MEG 2443']  # bads
+
+        # picks MEG gradiometers
+        picks = pick_types(raw.info, meg='grad', eeg=False, eog=True,
+                           stim=False, include=include, exclude='bads')
+
+        events = read_events(event_fname)
+
+        epochs = Epochs(raw, events[:10], event_id, tmin, tmax, picks=picks,
+                        baseline=(None, 0),
+                        reject=dict(grad=4000e-13, eog=150e-6), proj=False,
+                        preload=True)
+
+        tmin_full, tmax_full = -1, 1
+        epochs_full = Epochs(raw, events[:10], event_id, tmax=tmax_full,
+                             tmin=tmin_full, picks=picks,
+                             baseline=(None, 0),
+                             reject=dict(grad=4000e-13, eog=150e-6),
+                             proj=False, preload=True)
+
+        picks = pick_types(epochs.info, meg='grad', eeg=False, eog=True,
+                           stim=False, include=include, exclude='bads')
+        psds, freqs = compute_epochs_psd(epochs[:1], fmin=2, fmax=300,
+                                         n_fft=n_fft, picks=picks)
+
+        psds_t, freqs_t = compute_epochs_psd(epochs_full[:1], fmin=2, fmax=300,
+                                             tmin=tmin, tmax=tmax,
+                                             n_fft=n_fft, picks=picks)
+        # this one will fail if you add for example 0.1 to tmin
+        assert_array_almost_equal(psds, psds_t, 27)
+
+        psds_proj, _ = compute_epochs_psd(epochs[:1].apply_proj(), fmin=2,
+                                          fmax=300, n_fft=n_fft, picks=picks)
+
+        assert_array_almost_equal(psds, psds_proj)
+        assert_true(psds.shape == (1, len(picks), len(freqs)))
+        assert_true(np.sum(freqs < 0) == 0)
+        assert_true(np.sum(psds < 0) == 0)
```

Comment:
the same 4 lines yet again. Also, you have to do `picks` again etc creating many new lines which don't contribute to business logic. I feel this should also just go in with the other testing function.